### PR TITLE
feat: csharp enum rendering associated values

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR adds an extension to enums, which allows the retrieval of the associated enum values.

**Related issue(s)**
fixes https://github.com/asyncapi/modelina/issues/336

### CODE REVIEW

**Comment by jonaslagoni** [comment]
These functions are primarily for #356, however, I feel they are needed in the standard enum generation 🤔 Let me know if you disagree!

File: test/generators/csharp/__snapshots__/CSharpGenerator.spec.ts.snap [hunk 1]
```diff
@@ -191,25 +191,129 @@ exports[`CSharpGenerator should render \`class\` type 2`] = `
...
-}"
+}
+public static class StatesExtensions {
+  public static dynamic GetValue(this States enumValue)
+  {
+    switch (enumValue)
+    {
+      case States.Texas: return \\"Texas\\";
+      case States.Alabama: return \\"Alabama\\";
+      case States.California: return \\"California\\";
```
Comment (magicmatatjahu) [inline test/generators/csharp/__snapshots__/CSharpGenerator.spec.ts.snap:202]: I think that there is a bug. If something is a string, then it should be treated as native string in language, not as wrapped string with `""` characters. I don't know why, but it is rendered as object, not as string - please check also another test cases - you have this bug in every place.

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

File: src/generators/csharp/renderers/EnumRenderer.ts [hunk 2]
```diff
@@ -9,13 +9,34 @@ import { pascalCase } from 'change-case';
...
+    const enumItems = await this.renderItems();
     const formattedName = this.nameType(this.model.$id);
+    const getValueCaseItemValues = await this.getValueCaseItemValues();
+    const toEnumCaseItemValues = await this.toEnumCaseItemValues();
     return `public enum ${formattedName} {
-${this.indent(this.renderBlock(content, 2))}
-}`;
+${this.indent(enumItems)}
+}
+public static class ${formattedName}Extensions {
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply src/generators/csharp/renderers/EnumRenderer.ts:19]: It is a pattern in C#, not changeable 😄

File: test/generators/csharp/__snapshots__/CSharpGenerator.spec.ts.snap (hunk 1 above)
Comment (jonaslagoni) [reply test/generators/csharp/__snapshots__/CSharpGenerator.spec.ts.snap:202]: It is just the snapshot :) They are rendered as: 
```
      case States.Texas: return "Texas";
      case States.Alabama: return "Alabama";
      case States.California: return "California";
```
Which is correct.

File: src/generators/csharp/renderers/EnumRenderer.ts [hunk 3]
```diff
@@ -9,13 +9,34 @@ import { pascalCase } from 'change-case';
...
+  public static dynamic GetValue(this ${formattedName} enumValue)
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply src/generators/csharp/renderers/EnumRenderer.ts:20]: I know it is a weird format, but 🤷 - https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods

File: test/generators/csharp/__snapshots__/CSharpGenerator.spec.ts.snap (hunk 1 above)
Comment (magicmatatjahu) [reply test/generators/csharp/__snapshots__/CSharpGenerator.spec.ts.snap:202]: and it's one of the cases, why I don't like snapshots 😄

Comment (jonaslagoni) [reply test/generators/csharp/__snapshots__/CSharpGenerator.spec.ts.snap:202]: You just gotta get used to it 😆

File: src/generators/csharp/renderers/EnumRenderer.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply src/generators/csharp/renderers/EnumRenderer.ts:20]: Ok, it's something like trait.

**Review by magicmatatjahu — APPROVED** [review]

**Comment by jonaslagoni** [comment]
@magicmatatjahu `fix` or `feat`? 🤔 Kinda thinking it should be `feat`

**Pull request merged**